### PR TITLE
[W3M] SDK Interface cleanup

### PR DESCRIPTION
**Pull request by radeknovis**

### CODE REVIEW

File: Sources/Web3Modal/Web3ModalClient.swift [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            requiredNamespaces: Web3Modal.config.sessionParams.requiredNamespaces,
+            optionalNamespaces: Web3Modal.config.sessionParams.optionalNamespaces,
+            sessionProperties: Web3Modal.config.sessionParams.sessionProperties,
+            topic: topic
+        )
+    }
+    
+    /// For proposing a session to a wallet.
+    /// Function will propose a session on newly created pairing.
+    public func createPairingAndConnect() async throws -> WalletConnectURI {
```
Comment (llbartekll) [inline Sources/Web3Modal/Web3ModalClient.swift:82]: would make sense to call the method just connect() with an optional topic?
same as deprecated method of sign client

```
    public func connect(
        requiredNamespaces: [String: ProposalNamespace],
        optionalNamespaces: [String: ProposalNamespace]? = nil,
        sessionProperties: [String: String]? = nil,
        topic: String? = nil
```

Comment (llbartekll) [reply Sources/Web3Modal/Web3ModalClient.swift:82]: actually would match kotlin's interface too

File: Sources/Web3Modal/Web3Modal.swift [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        metadata: AppMetadata,
+        sessionParams: SessionParams = .default
+    ) {
+        Pair.configure(metadata: metadata)
+        Web3Modal.config = Web3Modal.Config(projectId: projectId, sessionParams: sessionParams)
+    }
+    
+    public static func set(sessionParams: SessionParams) {
+        Web3Modal.config.sessionParams = sessionParams
+    }
```
Comment (llbartekll) [inline Sources/Web3Modal/Web3Modal.swift:56]: what the set method is for?
what's the difference between configure()?

**Review by llbartekll — APPROVED** [review]

File: Sources/Web3Modal/Web3Modal.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Web3Modal/Web3Modal.swift:56]: I see, does it work? It’s singletons and config is used on the instantiation, so with the first call only

Comment (radeknovis) [reply Sources/Web3Modal/Web3Modal.swift:56]: Yes, it works. Config is stored as a static variable, which can be changed only via these 2 methods. I'm calling `set()` from Dapp

**Pull request merged**